drivers/stepper/adi_tmc/adi_tmc5041_stepper_controller.c

```diff
@@ -482,11 +474,12 @@ static int tmc5041_stepper_enable_constant_velocity_mode(const struct device *de
 {
 	LOG_DBG("Stepper motor controller %s enable constant velocity mode", dev->name);
 	const struct tmc5041_stepper_config *config = dev->config;
+	const struct tmc5041_config *tmc5041_config = config->controller->config;
 	struct tmc5041_stepper_data *data = dev->data;
-	uint32_t velocity_fclk;
 	int err;
 
-	calculate_velocity_from_hz_to_fclk(config->controller, velocity, &velocity_fclk);
+	uint32_t velocity_fclk = tmc5xxx_calculate_velocity_from_hz_to_fclk(velocity,
+								tmc5041_config->clock_frequency);
```

Comment:
the indentation here can be helped by keeping the definition of `uint32_t velocity_fclk` in the header (where it should be according to MISRA), and even defining tmc5041_config->clock_frequency:
```
uint32_t clock_frequency = tmc5041_config->clock_frequency;
uint32_t velocity_fclk;
...
velocity_fclk = tmc5xxx_calculate_velocity_from_hz_to_fclk(velocity, clock_frequency);
```
the compiler will optimize the `clock_frequency` away :)